utils/tests_fetcher.py

```diff
@@ -254,6 +256,122 @@ def diff_contains_doc_examples(repo: Repo, branching_point: str, filename: str)
     return old_content_clean != new_content_clean
 
 
+def get_impacted_files_from_tiny_model_summary(diff_with_last_commit: bool = False) -> List[str]:
+    """
+    Return a list of python modeling files that are impacted by the changes of `tiny_model_summary.json` in between:
+
+    - the current head and the main branch if `diff_with_last_commit=False` (default)
+    - the current head and its parent commit otherwise.
+
+    Returns:
+        `List[str]`: The list of Python modeling files that are impacted by the changes of `tiny_model_summary.json`.
+    """
+    repo = Repo(PATH_TO_REPO)
+
+    folder = Path(repo.working_dir)
+
+    if not diff_with_last_commit:
+        print(f"main is at {repo.refs.main.commit}")
+        print(f"Current head is at {repo.head.commit}")
+
+        commits = repo.merge_base(repo.refs.main, repo.head)
+        for commit in commits:
+            print(f"Branching commit: {commit}")
+    else:
+        print(f"main is at {repo.head.commit}")
+        commits = repo.head.commit.parents
+        for commit in commits:
+            print(f"Parent commit: {commit}")
+
+    if not os.path.isfile(folder / "tests/utils/tiny_model_summary.json"):
+        return []
+
+    files = set()
+    for commit in commits:
+        with checkout_commit(repo, commit):
+            with open(folder / "tests/utils/tiny_model_summary.json", "r", encoding="utf-8") as f:
+                old_content = f.read()
+
+        with open(folder / "tests/utils/tiny_model_summary.json", "r", encoding="utf-8") as f:
+            new_content = f.read()
+
+        # get the content as json object
+        old_content = json.loads(old_content)
+        new_content = json.loads(new_content)
+
+        old_keys = set(old_content.keys())
+        new_keys = set(new_content.keys())
```

Comment:
keys are always a set

Reply:
No. `'dict_keys'` is not the same as `set`: ` 'dict_keys' object has no attribute 'intersection'`

```python
a = {1: 2, 2: 3}
b = {2: 3, 3: 4}

a.keys()
b.keys()

c1 = set(a.keys()).intersection(set(b.keys()))
print(c1)


c2 = a.keys().intersection(b.keys())
print(c2)
```

gives

```
{2}
Traceback (most recent call last):
  File "C:\Users\33611\Desktop\Project\transformers\temp3.py", line 11, in <module>
    c2 = a.keys().intersection(b.keys())
AttributeError: 'dict_keys' object has no attribute 'intersection'

Process finished with exit code 1
```

Reply:
Alright! Thanks